File: examples/perl/regex.pl
#!/usr/bin/perl

# regex.pl - demonstrates some simple Perl regular expressions

# pulls IP addresses from stdin
#   note: not 100% correct; will take *any* digits separated by dots...
while(<>){
    $_ =~ m/(\d+\.\d+\.\d+\.\d+)/;
    if($1){
	print "IP: ".$1."\n";
    }
}

File: examples/python/regex.py
#!/usr/bin/python

'''some simple examples of regex use in Python'''

# import the regex package
import re

# compile your regex
p = re.compile('([a-z]+)(\d+)')

# now try to match it
m = p.match('abc123def456')

# matching in groups
print m.group(1)
print m.group(2)
